etc/profile-m-z/virt-manager.profile

```diff
@@ -0,0 +1,72 @@
+# Firejail profile for virt-manager
+# Description: Manage virtual machines
+# This file is overwritten after every install/update
+# Persistent local customizations
+include virt-manager.local
+# Persistent global definitions
+include globals.local
+
+noblacklist ${HOME}/.cache/virt-manager
+noblacklist ${RUNUSER}/libvirt
+
+noblacklist /sbin
+noblacklist /usr/sbin
+
+# Allow python 3 (blacklisted by disable-interpreters.inc)
+include allow-python3.inc
+
+include disable-common.inc
+include disable-devel.inc
+include disable-exec.inc
+include disable-interpreters.inc
+# breaks app
+#include disable-proc.inc
+include disable-programs.inc
+include disable-xdg.inc
+
+mkdir ${HOME}/.cache/virt-manager
+whitelist ${DOWNLOADS}
+whitelist ${HOME}/.cache/virt-manager
+whitelist ${RUNUSER}/libvirt
+whitelist /run/libvirt
+
+whitelist /usr/share/libvirt
+whitelist /usr/share/osinfo
+whitelist /usr/share/qemu
+whitelist /usr/share/seabios
+whitelist /usr/share/virt-manager
+# /usr/share/misc/usb.ids is a symlink to /var/lib/usbutils/usb.ids on Ubuntu 22.04
+whitelist /var/lib/usbutils/usb.ids
```

Comment:
```suggestion
```

Same as https://github.com/netblue30/firejail/pull/6226#discussion_r1505063245

Reply:
Good question indeed. While testing the profile I was surprised to see virt-manager break without it:

```sh
$ firejail /usr/bin/virt/manager
...

Error launching details: g-io-error-quark: Error opening file /usr/share/misc/usb.ids: No such file or directory (1)
Traceback (most recent call last):
  File "/usr/share/virt-manager/virtManager/vmwindow.py", line 40,
  in get_instance
      cls._instances[key] = vmmVMWindow(vm)
  File "/usr/share/virt-manager/virtManager/vmwindow.py", line 82,
  in __init__
      self._details = vmmDetails(self.vm, self.builder, self.topwin,
  File "/usr/share/virt-manager/virtManager/details/details.py",
line 408, in __init__
      self._init_details()
  File "/usr/share/virt-manager/virtManager/details/details.py",
line 714, in _init_details
    self._os_list = vmmOSList()
  File "/usr/share/virt-manager/virtManager/oslist.py", line 44, in
__init__
    self._init_state()
  File "/usr/share/virt-manager/virtManager/oslist.py", line 60, in
_init_state
    all_os = virtinst.OSDB.list_os(sortkey="label")
  File "/usr/share/virt-manager/virtinst/osdict.py", line 158, in
list_os
    self._os_db.get_os_list().get_elements()]
  File "/usr/share/virt-manager/virtinst/osdict.py", line 86, in
_os_db
    return self._os_loader.get_db()
  File "/usr/share/virt-manager/virtinst/osdict.py", line 79, in
_os_loader
    loader.process_default_path()
gi.repository.GLib.GError: g-io-error-quark: Error opening file /usr/
share/misc/usb.ids: No such file or directory (1)
```

So yes, the whitelisting _is_ needed here, at least for Ubuntu... Not sure if the whitelisting following symlinks is broken, haven't found the time yet to look into that.

HTH

Reply:
> Good question indeed. While testing the profile I was surprised to see
> virt-manager break without it:
> 
> ```shell
> $ firejail /usr/bin/virt/manager
> ...
> 
> Error launching details: g-io-error-quark: Error opening file /usr/share/misc/usb.ids: No such file or directory (1)
> Traceback (most recent call last):
> [...]
> ```
> 
> So yes, the whitelisting _is_ needed here, at least for Ubuntu... Not sure if
> the whitelisting following symlinks is broken, haven't found the time yet to
> look into that.

Strange, it looks like it did not find even the symlink itself.

Could it be that there are other paths being whitelisted in `/usr/share/misc`?

Or if it did find the symlink itself but not the target, maybe whitelisting
does not follow symlinks across mount paths.

What is the output of:

```sh
ls -l /usr/share/misc /var/lib/usbutils
firejail --profile=virt-manager ls -l /usr/share/misc /var/lib/usbutils
```

Misc: /usr/share/misc is present on Artix, but it's empty:

```sh
$ pacman -Qo /usr/share/misc
/usr/share/misc/ is owned by filesystem 2024.01.19-1
```

(To be clear, I don't consider this to be a blocker)

Reply:
```sh
$ ls -l /usr/share/misc /var/lib/usbutils
/usr/share/misc:
total 1360
-rwxr-xr-x 1 root root   49482 Jan 31  2022 config.guess
-rwxr-xr-x 1 root root   35406 Jan 31  2022 config.sub
lrwxrwxrwx 1 root root      13 Jan 26 06:26 magic -> ../file/magic
lrwxrwxrwx 1 root root      24 Sep 11 19:59 magic.mgc -> ../../lib/file/magic.mgc
-rw-r--r-- 1 root root 1299875 Jan 22  2022 pci.ids
lrwxrwxrwx 1 root root      25 Jan 26 06:26 usb.ids -> /var/lib/usbutils/usb.ids

/var/lib/usbutils:
total 700
-rw-r--r-- 1 root root 716218 Apr  2  2022 usb.ids

```

[whitelist /var/lib/usbutils/usb.ids]

```sh
$ firejail --profile=virt-manager ls -l /usr/share/misc /var/lib/usbutils
/usr/share/misc:
total 1360
-rwxr-xr-x 1 root root   49482 Jan 31  2022 config.guess
-rwxr-xr-x 1 root root   35406 Jan 31  2022 config.sub
lrwxrwxrwx 1 root root      13 Jan 26 06:26 magic -> ../file/magic
lrwxrwxrwx 1 root root      24 Sep 11 19:59 magic.mgc -> ../../lib/file/magic.mgc
-rw-r--r-- 1 root root 1299875 Jan 22  2022 pci.ids
lrwxrwxrwx 1 root root      25 Jan 26 06:26 usb.ids -> /var/lib/usbutils/usb.ids

/var/lib/usbutils:
total 700
-rw-r--r-- 1 root root 716218 Apr  2  2022 usb.ids

```

[#whitelist /var/lib/usbutils/usb.ids]

```sh
$ firejail --profile=virt-manager ls -l /usr/share/misc /var/lib/usbutils
/usr/share/misc:
total 1360
-rwxr-xr-x 1 root root   49482 Jan 31  2022 config.guess
-rwxr-xr-x 1 root root   35406 Jan 31  2022 config.sub
lrwxrwxrwx 1 root root      13 Jan 26 06:26 magic -> ../file/magic
lrwxrwxrwx 1 root root      24 Sep 11 19:59 magic.mgc -> ../../lib/file/magic.mgc
-rw-r--r-- 1 root root 1299875 Jan 22  2022 pci.ids
lrwxrwxrwx 1 root root      25 Jan 26 06:26 usb.ids -> /var/lib/usbutils/usb.ids

```

Nothing in any of my *.local files references /usr/share/misc on this Ubuntu box.

> (To be clear, I don't consider this to be a blocker)

That's quite allright. It's always better to look into these things before it ends up in master. Feel free to ask if there's anything else I can provide.